### PR TITLE
add workspace toolbar and workspace-dependency in chart service

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

**Comment by kretep** [comment]
Wouldn't it be better to refactor the whole workspace into a component and make the chart service a part of that? So that each workspace component manages its own charts? I think keeping the workspaces in the chart service as an array is spreading the logic too much.

**Comment by beaugogh** [comment]
test

**Comment by beaugogh** [comment]
My current understanding is this: refactoring the workspace (more precisely the cohort-selection part in the workspace, so let's call it cohort-selection-space) into a component is in ticket 205. And what you said is true in the sense that each cohort-selection-space is independent and reusable, with its own logics/scripts. But I think chart.service is another thing, it is sth that is and should be living outside any components. So if we want to create any logics/scripts that adhere to a specific component, we need to define directive(s) and/or controller(s) for that component, not services, and it is in ticket 205. I will look into it in that ticket, and see if I can move some of the functions in chart.service into the cohort-selection-space directive or controller. But before that, I think the current solution is a straightforward solution for this ticket, and once set, easy to use -- each time say we want to create a chart or filter a chart, we need to specify which cohort-selection-space that this chart belongs to, make sense, right?

**Comment by kretep** [comment]
Yes, I think the current changes are good for this PR. But I do think we need to refactor it again in ticket 205. Because I don't think the charts should be kept in a service to begin with, but in the component that contains the charts.

**Comment by beaugogh** [comment]
I agree :)

File: src/app/containers/workspace/cohort-selection/cohort-selection.html [hunk 1]
```diff
@@ -4,29 +4,56 @@
...
+                            uib-tooltip="Load cohorts"
+                            tooltip-trigger="mouseenter"
+                            tooltip-placement="top"
+                            popupDelay="1000"><span class="glyphicon glyphicon-cloud-download"></span></button>
+                    <button type="button"
+                            class="btn btn-secondary"
+                            uib-tooltip="Refresh"
+                            tooltip-trigger="mouseenter"
+                            tooltip-placement="top"
+                            popupDelay="1000"><span class="glyphicon glyphicon-refresh"></span></button>
```
Comment (kretep) [inline src/app/containers/workspace/cohort-selection/cohort-selection.html:33]: By the way, still the refresh button?

**Pull request merged**